asdf/treeutil.py

```diff
@@ -204,6 +204,19 @@ def __repr__(self):
 PendingValue = _PendingValue()
 
 
+class _RemoveNode:
```

Comment:
If someone wanted to remove a node, can't they just delete it from the tree explicitly? But this is presumably when an object is using its tag code to convert to ASDF and it wants ASDF to remove that attribute (or whatever) from its model, correct? But can't their tag code do that explicitly? But (Perry here is carrying on a discussion with himself) this is so the actual class code can signal this rather than having to go back and change the tag code? But code somewhere will have to be changed, right? [he heads to the psychiatrist for two personality disorder treatment...] This all arises from the presumption that a missing attribute will force a use of default, but we are discouraging that now, so this for some temporary mutant backward compatibility?

Reply:
That's a good point, I had assumed from the presence of this test:

https://github.com/asdf-format/asdf/blob/master/asdf/tests/test_api.py#L198

that somebody somewhere had a use case for removing nodes from the tree, but I haven't seen it yet.  It wouldn't be a tag class (like you mentioned, there are easier ways to do that) but `asdf.treeutil.walk_and_modify` is itself part of the public API so users can call it directly to process the tree and remove nodes.  That seems handy, but maybe we should leave out `RemoveNode` until someone actually needs it?

Reply:
Yeah, we may want to see who clamors for it first.

Reply:
Funny coincidence, I just needed this feature (though my task was a bit unusual).  I'm trying to compare schemas for validator content, so I needed to walk the schema tree and drop any of the annotation properties like description, title, etc.

Reply:
OK